src/lib/containers/discussion_forum/DiscussionReply.tsx

```diff
@@ -26,9 +32,20 @@ export const DiscussionReply: React.FC<DiscussionReplyProps> = ({
   const { accessToken } = useSynapseContext()
   const [message, setMessage] = useState<string>()
   const [showReplyModal, setShowReplyModal] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
   const { data: currentUserProfile } = useGetCurrentUserProfile()
+  const { data: moderatorList } = useGetModerators(reply.forumId)
+  const { mutate: deleteReply } = useDeleteReply({
+    onSuccess: () => {
+      setShowDeleteModal(false)
+      displayToast('A reply has been deleted.', 'info')
+    },
+  })
 
   const isAuthor = reply.createdBy == currentUserProfile?.ownerId
+  const isModerator = moderatorList?.results.includes(
+    currentUserProfile?.ownerId ?? '',
+  )
```

Comment:
Instead, fetch the entity bundle for the project and inspect `bundle.permissions.canModerate`.

Also, rename to `isCurrentUserModerator`. I think you might still need this moderator list to label the authors of threads and replies.

https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/auth/UserEntityPermissions.html